lib/pbench/test/unit/agent/task/test_tools_kill.py

```diff
@@ -281,17 +281,18 @@ def __init__(self, file_name: str, display_name: str):
         def mock_gen_run_directories(
             run_dir: pathlib.Path,
         ) -> List[Tuple[pathlib.Path, str]]:
-            return []
+            for run_dir in []:
+                yield run_dir, "uuid"
 
         runner = CliRunner(mix_stderr=False)
-        with (
-            mock.patch("pbench.cli.agent.commands.tools.kill.PidSource", MockPidSource),
-            mock.patch(
+        with mock.patch(
+            "pbench.cli.agent.commands.tools.kill.PidSource", MockPidSource
+        ):
+            with mock.patch(
                 "pbench.cli.agent.commands.tools.kill.gen_run_directories",
                 mock_gen_run_directories,
-            ),
-        ):
-            result = runner.invoke(kill.main)
+            ):
+                result = runner.invoke(kill.main)
```

Comment:
I thought you were going to replace `mock` with `monkeypatch`.

I believe that that fixture can be used as a (single) context, which would allow you to do multiple patches without nested `with` statements.

Ditto especially lines 359-370.  (I think Riya's recent PR has examples of how to do this.)

Reply:
I can't get rid of the `mock.patch` decorator use because `monkeypatch` does not offer that functionality, but everything else has been removed.